kubernetes-gcp-go/main.go

```diff
@@ -0,0 +1,152 @@
+			BinaryAuthorization: &container.ClusterBinaryAuthorizationArgs{
+				EvaluationMode: pulumi.String("PROJECT_SINGLETON_POLICY_ENFORCE"),
+			},
+			DatapathProvider: pulumi.String("ADVANCED_DATAPATH"),
+			Description:      pulumi.String("A GKE cluster"),
+			InitialNodeCount: pulumi.Int(1),
+			IpAllocationPolicy: &container.ClusterIpAllocationPolicyArgs{
+				ClusterIpv4CidrBlock:  pulumi.String("/14"),
+				ServicesIpv4CidrBlock: pulumi.String("/20"),
+			},
+			Location: pulumi.String(gcpRegion),
+			MasterAuthorizedNetworksConfig: &container.ClusterMasterAuthorizedNetworksConfigArgs{
+				CidrBlocks: container.ClusterMasterAuthorizedNetworksConfigCidrBlockArray{
+					&container.ClusterMasterAuthorizedNetworksConfigCidrBlockArgs{
+						CidrBlock:   pulumi.String("0.0.0.0/0"),
+						DisplayName: pulumi.String("All networks"),
+					},
+				},
+			},
+			Network:        gkeNetwork.Name,
+			NetworkingMode: pulumi.String("VPC_NATIVE"),
+			PrivateClusterConfig: &container.ClusterPrivateClusterConfigArgs{
+				EnablePrivateNodes:    pulumi.Bool(true),
+				EnablePrivateEndpoint: pulumi.Bool(false),
+				MasterIpv4CidrBlock:   pulumi.String("10.100.0.0/28"),
+			},
+			RemoveDefaultNodePool: pulumi.Bool(true),
+			ReleaseChannel: &container.ClusterReleaseChannelArgs{
+				Channel: pulumi.String("STABLE"),
+			},
+			Subnetwork: gkeSubnet.Name,
+			WorkloadIdentityConfig: &container.ClusterWorkloadIdentityConfigArgs{
+				WorkloadPool: pulumi.String(fmt.Sprintf("%v.svc.id.goog", gcpProject)),
+			},
+		})
+		if err != nil {
+			return err
+		}
+
+		// Create a GCP Service Account for the node pool
+		gkeNodepoolSa, err := serviceaccount.NewAccount(ctx, "gke-nodepool-sa", &serviceaccount.AccountArgs{
+			AccountId:   pulumi.String("nodepool-1-sa"),
+			DisplayName: pulumi.String("Nodepool 1 Service Account"),
+		})
+		if err != nil {
+			return err
+		}
+
+		// Create a new node pool
+		_, err = container.NewNodePool(ctx, "gke-nodepool", &container.NodePoolArgs{
+			Cluster:   gkeCluster.ID(),
+			NodeCount: pulumi.Int(nodePoolCount),
+			NodeConfig: &container.NodePoolNodeConfigArgs{
+				OauthScopes: pulumi.StringArray{
+					pulumi.String("https://www.googleapis.com/auth/cloud-platform"),
+				},
+				ServiceAccount: gkeNodepoolSa.Email,
+			},
+		})
+		if err != nil {
+			return err
+		}
+
+		// Build Kubeconfig for accessing the cluster
+		clusterKubeconfig := pulumi.Sprintf(`apiVersion: v1
+clusters:
+- cluster:
+    certificate-authority-data: %[3]s
+    server: https://%[2]s
+  name: %[1]s
+contexts:
+- context:
+    cluster: %[1]s
+    user: %[1]s
+  name: %[1]s
+current-context: %[1]s
+kind: Config
+preferences: {}
+users:
+- name: %[1]s
+  user:
+    exec:
+      apiVersion: client.authentication.k8s.io/v1beta1
+      command: gke-gcloud-auth-plugin
+      installHint: Install gke-gcloud-auth-plugin for use with kubectl by following
+        https://cloud.google.com/blog/products/containers-kubernetes/kubectl-auth-changes-in-gke
```

Comment:
Interesting -- how does this work?

Reply:
I have never actually seen it happen but i believe it's the error message that gets shown to a user if the don't have ```gke-gcloud-auth-plugin``` on the system.  Not actually sure how it will work in a Pulumi run though.

Reply:
I see this myself, yeah. I'm using `brew` to manage my `gcloud` binary, so it looks like `gke-gcloud-auth-plugin` gets installed in the same directory as `gcloud`, but because only `gcloud` is symlinked, it's is not on my path. When I tweak my path manually to include `gke-gcloud-auth-plugin`, though, the message goes away. 

Unfortunately the linked article doesn't address the Homebrew-installed use case, so I had to fumble around looking for `gke-gcloud-auth-plugin` on my system to figure it out. I'm assuming that probably means most people don't install `gcloud` with Brew.